File: Leetcode/897-IncreasingOrderSearchTree.cpp
class Solution {
public:
    TreeNode* increasingBST(TreeNode* root) {
        if (!root) {
            return nullptr;
        }

        TreeNode* dummy = new TreeNode(0);
        TreeNode* current = dummy;

        inorderTraversal(root, current);

        return dummy->right;
    }

private:
    void inorderTraversal(TreeNode* node, TreeNode*& current) {
        if (!node) {
            return;
        }

        inorderTraversal(node->left, current);

        node->left = nullptr;
        current->right = node;
        current = node;

        inorderTraversal(node->right, current);
    }
};

File: Leetcode/897-IncreasingOrderSearchTree.py
class Solution:
    def increasingBST(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
        dummy = TreeNode()
        k = dummy
        stack = []

        while stack or root:

            while root:
                stack.append(root)
                root = root.left
            root = stack.pop()
                
            k.right = root
            root.left = None
            k = root

            root = root.right

        return dummy.right
